File: src/java/eu/sig/training/ch04/v3/CheckingAccount.java
package eu.sig.training.ch04.v3;

import eu.sig.training.ch04.BusinessException;
import eu.sig.training.ch04.Money;

// tag::CheckingAccount[]
public class CheckingAccount extends Account {
    private int transferLimit = 100;

    @Override
    public Transfer makeTransfer(String counterAccount, Money amount)
        throws BusinessException {
        if (amount.greaterThan(this.transferLimit)) {
            throw new BusinessException("Limit exceeded!");
        }
        return super.makeTransfer(counterAccount, amount);
    }
}
// end::CheckingAccount[]

File: src/csharp/eu/sig/training/ch04/v3/CheckingAccount.cs
namespace eu.sig.training.ch04.v3
{

    // tag::CheckingAccount[]
    public class CheckingAccount : Account
    {
        private int transferLimit = 100;

        public override Transfer MakeTransfer(string counterAccount, Money amount)
        {
            if (amount.GreaterThan(this.transferLimit))
            {
                throw new BusinessException("Limit exceeded!");
            }
            return base.MakeTransfer(counterAccount, amount);
        }
    }
    // end::CheckingAccount[]

}